backend/dataall/modules/mlstudio/cdk/mlstudio_extension.py

```diff
@@ -31,83 +29,92 @@ def extent(setup: EnvironmentSetup):
-        cdk_look_up_role_arn = SessionHelper.get_cdk_look_up_role_arn(
-            accountid=_environment.AwsAccountId, region=_environment.region
-        )
-        existing_default_vpc = EC2.check_default_vpc_exists(
-            AwsAccountId=_environment.AwsAccountId, region=_environment.region, role=cdk_look_up_role_arn
-        )
-        if existing_default_vpc:
-            logger.info("Using default VPC for Sagemaker Studio domain")
-            # Use default VPC - initial configuration (to be migrated)
-            vpc = ec2.Vpc.from_lookup(setup, 'VPCStudio', is_default=True)
-            subnet_ids = [private_subnet.subnet_id for private_subnet in vpc.private_subnets]
-            subnet_ids += [public_subnet.subnet_id for public_subnet in vpc.public_subnets]
-            subnet_ids += [isolated_subnet.subnet_id for isolated_subnet in vpc.isolated_subnets]
+
+        if domain.vpcId and domain.subnetIds:
+            logger.info(f'Using VPC {domain.vpcId} and subnets {domain.subnetIds} for SageMaker Studio domain')
+            vpc = ec2.Vpc.from_lookup(setup, 'VPCStudio', vpc_id=domain.vpcId)
+            subnet_ids = domain.subnetIds
             security_groups = []
         else:
-            logger.info("Default VPC not found, Exception. Creating a VPC for SageMaker resources...")
-            # Create VPC with 3 Public Subnets and 3 Private subnets wit NAT Gateways
-            log_group = logs.LogGroup(
-                setup,
-                f'SageMakerStudio{_environment.name}',
-                log_group_name=f'/{_environment.resourcePrefix}/{_environment.name}/vpc/sagemakerstudio',
-                retention=logs.RetentionDays.ONE_MONTH,
-                removal_policy=RemovalPolicy.DESTROY,
-            )
-            vpc_flow_role = iam.Role(
-                setup, 'FlowLog',
-                assumed_by=iam.ServicePrincipal('vpc-flow-logs.amazonaws.com')
-            )
-            vpc = ec2.Vpc(
-                setup,
-                "SageMakerVPC",
-                max_azs=3,
-                cidr="10.10.0.0/16",
-                subnet_configuration=[
-                    ec2.SubnetConfiguration(
-                        subnet_type=ec2.SubnetType.PUBLIC,
-                        name="Public",
-                        cidr_mask=24
-                    ),
-                    ec2.SubnetConfiguration(
-                        subnet_type=ec2.SubnetType.PRIVATE_WITH_NAT,
-                        name="Private",
-                        cidr_mask=24
-                    ),
-                ],
-                enable_dns_hostnames=True,
-                enable_dns_support=True,
-            )
-            ec2.FlowLog(
-                setup, "StudioVPCFlowLog",
-                resource_type=ec2.FlowLogResourceType.from_vpc(vpc),
-                destination=ec2.FlowLogDestination.to_cloud_watch_logs(log_group, vpc_flow_role)
+            cdk_look_up_role_arn = SessionHelper.get_cdk_look_up_role_arn(
```

Comment:
Since we already classified the VPC as "imported", "created" and "default", i think we can remove this block and reuse the domain info right?

Reply:
Im not so sure about that because what about for backwards compatibility...

Right now I classify the backwards compatible MLStudioDomains already created as type `unknown` because I did not know of a better way to populate the sagemaker studio domain RDS Table with `vpcType` by looking into each environment and checking the domain's networking configuration